pkg/apis/core/validation/validation.go

```diff
@@ -638,6 +638,16 @@ func validateVolumeSource(source *core.VolumeSource, fldPath *field.Path, volNam
 			allErrs = append(allErrs, validateScaleIOVolumeSource(source.ScaleIO, fldPath.Child("scaleIO"))...)
 		}
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.CSIInlineVolume) {
+		if source.CSI != nil {
+			if numVolumes > 0 {
+				allErrs = append(allErrs, field.Forbidden(fldPath.Child("csi"), "may not specify more than 1 volume type"))
+			} else {
+				numVolumes++
+				allErrs = append(allErrs, validateCSIVolumeSource(source.CSI, fldPath.Child("csi"))...)
+			}
+		}
+	}
```

Comment:
If feature is disabled, should validation reject the object? Accepting it without validation seems like a bad idea.

Reply:
@saad-ali not quite following you. 

If feature is enabled, only then the object is validated because when the feature is not enabled, the object (`VolumeSource.CSI`) is cleared as nil (see https://github.com/kubernetes/kubernetes/blob/19be94c88bec6b3cd5983e3f7a468b1e7b30c8bb/pkg/api/pod/util.go#L342)

Reply:
that is correct, but as described in https://github.com/kubernetes/kubernetes/pull/67452#discussion_r215694746, also checking in validation ensured that if the PrepareForCreate/PrepareForUpdate check is removed or skipped, validation prevents alpha data from being persisted when the feature is not enabled.

Reply:
Just in case the field drop doesn't work, it would be good to return an error in validation if CSI is also set and the feature is off.

Reply:
@liggitt I misread @saad-ali concerns.  Fixed cc @msau42 

PTAL